lix/source-code/client/src/openRepository.ts

```diff
@@ -24,56 +24,80 @@ export async function openRepository(
 	args: {
 		nodeishFs: NodeishFilesystem
 		workingDirectory?: string
-		corsProxy?: string
 		auth?: unknown // unimplemented
 	}
 ): Promise<Repository> {
 	const rawFs = args.nodeishFs
 
 	const [errors, setErrors] = createSignal<Error[]>([])
 
-	// parse url in the format of github.com/inlang/example and split it to host, owner and repo
-	const [host, owner, repoName] = [...url.split("/")]
+	// the url format for lix urls is
+	// https://lix.inlang.com/git/github.com/inlang/monorepo
+	// proto:// lixDomain / namespace / repoHost / owner / repoName
+	// namespace is ignored until switching from git.inlang.com to lix.inlang.com and can eveolve in future to be used for repoType, api type or feature group
+	// the url format for direct github urls without a lix server is https://github.com/inlang/examplX (only per domain-enabled git hosters will be supported, currently just gitub)
+	// the url for opening a local repo allready in the fs provider is file://path/to/repo (not implemented yet)
+
+	const { origin, pathname } = new URL(url)
+	let gitProxyUrl = ""
+	let gitHubProxyUrl = ""
+
+	let namespace = ""
+	let host = ""
+	let owner = ""
+	let repoName = ""
+
+	if (origin === "https://github.com") {
+		// parse url in the format of github.com/inlang/example and split it to host, owner and repo
+		const pathParts = pathname.split("/")
+
+		host = "github.com"
+		owner = pathParts[1] || ""
+		repoName = pathParts[2] || ""
+		// check if all 3 parts are present, if not, return an error
+		if (!host || !owner || !repoName) {
+			throw new Error(
+				`Invalid url format for '${url}' for direct cloning repository from github, please use the format of https://github.com/inlang/monorepo.`
+			)
+		}
+	} else {
+		const pathParts = pathname.split("/")
 
-	// check if all 3 parts are present, if not, return an error
-	if (!host || !owner || !repoName) {
-		throw new Error(
-			`Invalid url format for '${url}' for cloning repository, please use the format of github.com/inlang/example.`
-		)
-	}
-	if (args.auth) {
-		console.warn("Auth currently not implemented in lisa client")
-	}
+		gitProxyUrl = origin + "/git-proxy/"
+		gitHubProxyUrl = origin + "/github-proxy/"
 
-	let gitHubProxyBaseUrl = ""
-	const gitProxyUrl = args?.corsProxy
+		namespace = pathParts[1] || ""
+		host = pathParts[2] || ""
+		owner = pathParts[3] || ""
+		repoName = pathParts[4] || ""
 
-	// Git cors proxy and github cors proxy have to be on the same server and will be unified further
-	if (gitProxyUrl && gitProxyUrl?.startsWith("http")) {
-		const { origin } = new URL(gitProxyUrl)
-		gitHubProxyBaseUrl = origin
+		// check if all 3 parts are present, if not, return an error
+		if (!namespace || !host || !owner || !repoName) {
+			throw new Error(
+				`Invalid url format for '${url}' for cloning repository, please use the format of https://lix.inlang.com/git/github.com/inlang/monorepo.`
+			)
```

Comment:
Should we write a function `isValidLixUri` that we can use to a) verify URIs and b) write tests for? 

I am referring to URI here because lix will be able to open local repositories where the path is not a URL but an URI.